### PR TITLE
Asm

### PR DESCRIPTION
**Pull request by lukasj**
relates to #1051 in master:
* removes eclipselink asm from the main build
* main build treats eclipselink asm as any other other library

### CODE REVIEW

**Comment by rfelcman** [comment]
Just a few comments:
Travis-CI fix: maybe add to `.travis.yml` before line 79 ASM build to populate Maven local repo.

File: plugins/org.eclipse.persistence.asm/pom.xml [hunk 1]
```diff
@@ -110,6 +253,63 @@
...
+                            <!-- ASM itself uses 1.8 source level and then retrofits the bytecode
+                            to work on Java SE 5+. We don't use retrofitter, so we need
+                            to fix sources to compile for Java SE 7 (EclipseLink 2.6.x) -->
+                            <target>
+                                <replace file="${project.build.directory}/generated-sources/asm-src/org/eclipse/persistence/internal/libraries/asm/tree/Util.java">
+                                    <replacefilter>
+                                        <replacetoken><![CDATA[new ArrayList<>(1)]]></replacetoken>
+                                        <replacevalue><![CDATA[new ArrayList<T>(1)]]></replacevalue>
+                                    </replacefilter>
+                                </replace>
```
Comment (rfelcman) [inline plugins/org.eclipse.persistence.asm/pom.xml:276]: Interesting I believe the best solution from possible options, but I think, that should be applied to all *.java files in `generated-sources` to cover possible ASM code changes.

Comment (lukasj) [reply plugins/org.eclipse.persistence.asm/pom.xml:276]: that would have 2 drawbacks:
* it's faster to change 1 file than XX files
* it's probably better to "track" necessary changes, git won't do it for us here

File: plugins/org.eclipse.persistence.asm/pom.xml [hunk 2]
```diff
@@ -18,17 +18,106 @@
          xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance">
     <modelVersion>4.0.0</modelVersion>
 
-    <name>EclipseLink ASM</name>
+    <parent>
```
Comment (rfelcman) [inline plugins/org.eclipse.persistence.asm/pom.xml:21]: Line 4: Copyright Year

File: pom.xml [hunk 3]
```diff
@@ -161,6 +161,8 @@
         <test.xml.parser>org.eclipse.persistence.platform.xml.jaxp.JAXPParser</test.xml.parser>
 
         <!--Eclipse Dependencies version-->
+        <!-- TODO: FIX ME before merge-->
```
Comment (rfelcman) [inline pom.xml:164]: TODO fix?

File: plugins/org.eclipse.persistence.asm/src/main/java/org/eclipse/persistence/internal/libraries/asm/EclipseLinkASMClassWriter.java [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+                // ie SecurityException
+                LOG.log(Level.WARNING, "Cannot read 'java.specification.version' property.", t);
+                if (LOG.isLoggable(Level.FINE)) {
+                    LOG.log(Level.FINE, "Generating bytecode for Java SE ''{0}''.", v);
+                }
+            }
+        }
+        Integer version = versionMap.get(v);
+        if (version == null) {
+            // current JDK is iether too new
```
Comment (rfelcman) [inline plugins/org.eclipse.persistence.asm/src/main/java/org/eclipse/persistence/internal/libraries/asm/EclipseLinkASMClassWriter.java:103]: iether -> either

**Review by rfelcman — CHANGES_REQUESTED** [review]
Almost OK except mentioned minor comments.
Two things which should be optionally included there:

- New Travis-CI target something like

```
...
  matrix:
    - MAVEN_TEST=asm
...
  - if [ "$MAVEN_TEST" == "asm" ]; then
      cd plugins/org.eclipse.persistence.asm/;
      mvn test;
    fi
```

- Maybe make generated Javadoc little bit nicer (title, copyright year)

File: plugins/org.eclipse.persistence.asm/src/main/java/org/eclipse/persistence/internal/libraries/asm/EclipseLinkASMClassWriter.java (hunk 4 above)
Comment (lukasj) [reply plugins/org.eclipse.persistence.asm/src/main/java/org/eclipse/persistence/internal/libraries/asm/EclipseLinkASMClassWriter.java]: done

File: plugins/org.eclipse.persistence.asm/pom.xml (hunk 2 above)
Comment (lukasj) [reply plugins/org.eclipse.persistence.asm/pom.xml:21]: done

File: pom.xml (hunk 3 above)
Comment (lukasj) [reply pom.xml:164]: this is intentional to trigger extra build/tests for "final" asm build once it's ready in staging to double check everything is fine

**Comment by lukasj** [comment]
wrt travis for asm - it is non-trivial to add it correctly - the one built by travis won't be the one used by main eclispelink build there without extra modification of eclipselink's pom - I don't think we want to be doing that

wrt javadoc - I can see correct cp year in the javadoc footer; I don't think we need to customize the default more

**Review by rfelcman — APPROVED** [review]
LGTM

**Pull request merged**